<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1222</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ngl Qt Demo</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="s_mainWindowGridLayout">
    <item row="0" column="0" rowspan="2">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>285</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="m_boidTab">
       <attribute name="title">
        <string>Boid</string>
       </attribute>
       <widget class="QGroupBox" name="s_boidParams">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>289</width>
          <height>161</height>
         </rect>
        </property>
        <property name="title">
         <string>BoidSetup</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Boid Count</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="m_boidCount">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>3000</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="boidSpeed">
           <property name="maximum">
            <double>3.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>maxSpeed</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="predatorAdd">
           <property name="text">
            <string>Add Predator</string>
           </property>
           <property name="iconSize">
            <size>
             <width>8</width>
             <height>8</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>add Predator</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Prey Flee Factor</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="m_preyFleeFactor">
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QLabel" name="label_18">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>160</y>
          <width>141</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Predator Seek Factor</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="m_predatorSeekFactor">
        <property name="geometry">
         <rect>
          <x>175</x>
          <y>160</y>
          <width>91</width>
          <height>31</height>
         </rect>
        </property>
        <property name="maximum">
         <double>99.099999999999994</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="m_flockTab">
       <attribute name="title">
        <string>Flock</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>289</width>
          <height>141</height>
         </rect>
        </property>
        <property name="title">
         <string>FlockBehaviour</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="2">
          <widget class="QDoubleSpinBox" name="m_keepApartRange"/>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Keep Apart Dist: </string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Alignment</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Separation</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QDoubleSpinBox" name="separationFactor">
           <property name="minimum">
            <double>-20.000000000000000</double>
           </property>
           <property name="maximum">
            <double>20.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QDoubleSpinBox" name="alignFactor">
           <property name="minimum">
            <double>-20.000000000000000</double>
           </property>
           <property name="maximum">
            <double>20.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="cohesionFactor">
           <property name="minimum">
            <double>-20.000000000000000</double>
           </property>
           <property name="maximum">
            <double>20.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Cohesion</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QDoubleSpinBox" name="m_seekRange"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Seek Range:</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="BoidDraw">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>180</y>
          <width>285</width>
          <height>91</height>
         </rect>
        </property>
        <property name="title">
         <string>Draw</string>
        </property>
        <widget class="QLabel" name="label_9">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>60</y>
           <width>91</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Pick Shader</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>30</y>
           <width>79</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Pick Mesh</string>
         </property>
        </widget>
        <widget class="QComboBox" name="boidShapeType">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>20</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>31</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Ubuntu</family>
          </font>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>Mesh</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sphere</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Tri</string>
          </property>
         </item>
        </widget>
        <widget class="QComboBox" name="m_shaderName">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>60</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>Phong</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Constant</string>
          </property>
         </item>
        </widget>
        <widget class="QFrame" name="frame">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>20</y>
           <width>261</width>
           <height>31</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
        </widget>
        <widget class="QFrame" name="frame_2">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>60</y>
           <width>261</width>
           <height>31</height>
          </rect>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
        </widget>
        <zorder>frame</zorder>
        <zorder>frame_2</zorder>
        <zorder>label_9</zorder>
        <zorder>label_8</zorder>
        <zorder>boidShapeType</zorder>
        <zorder>m_shaderName</zorder>
       </widget>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>290</y>
          <width>281</width>
          <height>121</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Neighbour Parameters</string>
        </property>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>40</y>
           <width>211</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Neighbour Search Radius:</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_12">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>90</y>
           <width>191</width>
           <height>21</height>
          </rect>
         </property>
         <property name="toolTip">
          <string>larger values improves behaviour but slows down simulation</string>
         </property>
         <property name="text">
          <string>Neighbours Consider %</string>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="m_neighbourRegionRadius">
         <property name="geometry">
          <rect>
           <x>210</x>
           <y>30</y>
           <width>66</width>
           <height>31</height>
          </rect>
         </property>
        </widget>
        <widget class="QSpinBox" name="m_neighboursConsider">
         <property name="geometry">
          <rect>
           <x>210</x>
           <y>80</y>
           <width>66</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>20</number>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="m_obstacleTab">
       <attribute name="title">
        <string>Obstacle</string>
       </attribute>
       <widget class="QGroupBox" name="obstacleSetup">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>10</y>
          <width>285</width>
          <height>101</height>
         </rect>
        </property>
        <property name="title">
         <string>Obstacle Setup</string>
        </property>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>30</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Count</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>70</y>
           <width>111</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>AvoidFactor</string>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="m_avoidFactor">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>60</y>
           <width>121</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.200000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
        <widget class="QSpinBox" name="m_obstacleCount">
         <property name="geometry">
          <rect>
           <x>155</x>
           <y>20</y>
           <width>111</width>
           <height>31</height>
          </rect>
         </property>
         <property name="maximum">
          <number>2</number>
         </property>
        </widget>
       </widget>
       <widget class="QDoubleSpinBox" name="m_obsAffectRadius">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>110</y>
          <width>91</width>
          <height>31</height>
         </rect>
        </property>
        <property name="minimum">
         <double>12.000000000000000</double>
        </property>
        <property name="maximum">
         <double>50.000000000000000</double>
        </property>
        <property name="value">
         <double>18.000000000000000</double>
        </property>
       </widget>
       <widget class="QLabel" name="label_19">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>120</y>
          <width>161</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Obstacle affect Radius</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Container</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>281</width>
          <height>121</height>
         </rect>
        </property>
        <property name="title">
         <string>Container Setup</string>
        </property>
        <widget class="QLabel" name="label_13">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>20</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>CONTAINER SIZE</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="m_sizex">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>61</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimum">
          <number>60</number>
         </property>
         <property name="maximum">
          <number>120</number>
         </property>
         <property name="singleStep">
          <number>20</number>
         </property>
         <property name="value">
          <number>80</number>
         </property>
        </widget>
        <widget class="QSpinBox" name="m_sizey">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>40</y>
           <width>66</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimum">
          <number>60</number>
         </property>
         <property name="maximum">
          <number>120</number>
         </property>
         <property name="singleStep">
          <number>20</number>
         </property>
         <property name="value">
          <number>80</number>
         </property>
        </widget>
        <widget class="QSpinBox" name="m_sizez">
         <property name="geometry">
          <rect>
           <x>180</x>
           <y>40</y>
           <width>66</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimum">
          <number>60</number>
         </property>
         <property name="maximum">
          <number>120</number>
         </property>
         <property name="singleStep">
          <number>20</number>
         </property>
         <property name="value">
          <number>80</number>
         </property>
        </widget>
        <widget class="QCheckBox" name="m_drawBBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>90</y>
           <width>161</width>
           <height>26</height>
          </rect>
         </property>
         <property name="text">
          <string>Draw Bounding Box</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>130</y>
          <width>271</width>
          <height>91</height>
         </rect>
        </property>
        <property name="title">
         <string>Container Avoidance</string>
        </property>
        <widget class="QLabel" name="label_14">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>151</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Wall Avoid Strength</string>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="m_wallAvoidStrength">
         <property name="geometry">
          <rect>
           <x>180</x>
           <y>10</y>
           <width>66</width>
           <height>31</height>
          </rect>
         </property>
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
         <property name="value">
          <double>2.000000000000000</double>
         </property>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1222</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
